pkg/controller/util.go

```diff
@@ -616,3 +624,62 @@ func SetPodPvcAnnotations(pod *v1.Pod, pvc *v1.PersistentVolumeClaim) {
 		}
 	}
 }
+
+// GetPreallocation retuns the preallocation setting for DV, falling back to StorageClass and global setting (in this order)
+func GetPreallocation(client client.Client, dataVolume *cdiv1.DataVolume) bool {
+	// First, the DV's preallocation
+	if dataVolume.Spec.Preallocation != nil {
+		return *dataVolume.Spec.Preallocation
+	}
+
+	cdiconfig := &cdiv1.CDIConfig{}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: common.ConfigName}, cdiconfig); err != nil {
+		klog.Errorf("Unable to find CDI configuration, %v\n", err)
+		return defaultPreallocation
+	}
+	if cdiconfig.Status.Preallocation == nil {
+		// No need to bother reading storage class if Preallocation not defined in the config
+		return defaultPreallocation
+	}
+
+	// Second, the config setting for the storage class
+	sc := GetStorageClassNameForDV(client, dataVolume)
+	if sc != "" {
+		if pre, exists := cdiconfig.Status.Preallocation.StorageClass[sc]; exists {
+			return pre
+		}
+	}
+
+	// Third, the global setting
+	if cdiconfig.Status.Preallocation.Global != nil {
+		return *cdiconfig.Status.Preallocation.Global
+	}
+
+	// Finally, the default false
+	return defaultPreallocation
+}
+
+// GetStorageClassNameForDV returns storage class to be used for the DV's PVC
+func GetStorageClassNameForDV(c client.Client, dv *cdiv1.DataVolume) string {
+	// If DV has a SC, return it
+	if dv != nil && dv.Spec.PVC != nil && dv.Spec.PVC.StorageClassName != nil && *dv.Spec.PVC.StorageClassName != "" {
+		return *dv.Spec.PVC.StorageClassName
+	}
+
+	// If DV's PVC has a SC, return it
+	pvc := &v1.PersistentVolumeClaim{}
+	// TODO change when PVC's name is different from DV's name
+	err := c.Get(context.TODO(), types.NamespacedName{Name: dv.Name, Namespace: dv.Namespace}, pvc)
```

Comment:
I see more comments like that: "// TODO change when PVC's name is different from DV's name", and I am guilty of adding at least one ;)
I suppose this calls for a function GetPvcForDv or something like that. Might be good candidate for a cleanup PR.

Reply:
I would qualify such a task a technical debt.